src/main/java/somethingrandom/entity/ReferenceItem.java

```diff
@@ -11,6 +11,10 @@ public class ReferenceItem extends Item {
      * The Name.
      */
     private String name;
```

Comment:
I think these will shadow the declarations in `Item`, so the constructor will probably not set any of the fields correctly.

Probably best way to fix this is to use `super` in the constructor to set those fields.

Other options: either `Item` has protected fields (instead of public/private), or `ReferenceItem` etc. only use its getters and setters.

<details><summary>Failing example test</summary>

```java
package somethingrandom.entity;

import org.junit.Test;

import java.time.Duration;
import java.time.Instant;

import static org.junit.Assert.*;

public class ActionableItemTest {
    @Test
    public void shouldUseConstructorParameters() {
        ActionableItem item = new ActionableItem("Name", Instant.MIN, Duration.ZERO);

        assertEquals("Name", item.getName());
    }
}
```
fails with:
```
java.lang.AssertionError: 
Expected :Name
Actual   :null
```
</details>